java/piranha/src/test/java/com/uber/piranha/XPFlagCleanerTest.java

```diff
@@ -2120,4 +2120,183 @@ public void exceptionWhenArgumentIndexNotSpecifiedAndNotArgumentIndexOptional()
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void testEmptyFlagRemovesAnnotatedMethodsTreatment() throws IOException {
+
+    ErrorProneFlags.Builder b = ErrorProneFlags.builder();
+    b.putFlag("Piranha:FlagName", "");
+    b.putFlag("Piranha:IsTreated", "true");
+    b.putFlag("Piranha:Config", "config/properties.json");
+
+    BugCheckerRefactoringTestHelper bcr =
+        BugCheckerRefactoringTestHelper.newInstance(new XPFlagCleaner(b.build()), getClass());
+
+    bcr = bcr.setArgs("-d", temporaryFolder.getRoot().getAbsolutePath());
+
+    bcr = addHelperClasses(bcr);
+    bcr.addInputLines(
+            "EmptyFlagRemovesAnnotatedMethods.java",
+            "package com.uber.piranha;",
+            "class EmptyFlagRemovesAnnotatedMethods {",
+            "  enum TestExperimentName {",
+            "    SAMPLE_STALE_FLAG",
+            "  }",
+            "  public String evaluate(int x) {",
+            "    if (x == 1) { return \"yes\"; }",
+            "    return \"no\";",
+            "  }",
+            "  @ToggleTesting(treated = TestExperimentName.SAMPLE_STALE_FLAG)",
+            "  public void x () {}",
+            "}")
+        .addOutputLines(
+            "EmptyFlagRemovesAnnotatedMethods.java",
+            "package com.uber.piranha;",
+            "class EmptyFlagRemovesAnnotatedMethods {",
+            "  enum TestExperimentName {",
+            "    SAMPLE_STALE_FLAG",
+            "  }",
+            "  public String evaluate(int x) {",
+            "    if (x == 1) { return \"yes\"; }",
+            "    return \"no\";",
+            "  }",
+            "  @ToggleTesting(treated = TestExperimentName.SAMPLE_STALE_FLAG)",
+            "  public void x () {}",
+            "}")
+        .addInputLines(
+            "ToggleTesting.java",
+            "package com.uber.piranha;",
+            "import java.lang.annotation.ElementType;",
+            "import java.lang.annotation.Retention;",
+            "import java.lang.annotation.RetentionPolicy;",
+            "import java.lang.annotation.Target;",
+            "@Retention(RetentionPolicy.RUNTIME)",
+            "@Target({ElementType.METHOD})",
+            "@interface ToggleTesting {",
+            "  EmptyFlagRemovesAnnotatedMethods.TestExperimentName treated();",
+            "}")
+        .addOutputLines(
+            "ToggleTesting.java",
+            "package com.uber.piranha;",
+            "import java.lang.annotation.ElementType;",
+            "import java.lang.annotation.Retention;",
+            "import java.lang.annotation.RetentionPolicy;",
+            "import java.lang.annotation.Target;",
+            "@Retention(RetentionPolicy.RUNTIME)",
+            "@Target({ElementType.METHOD})",
+            "@interface ToggleTesting {",
+            "  EmptyFlagRemovesAnnotatedMethods.TestExperimentName treated();",
+            "}")
+        .doTest();
+  }
+
+  @Test
+  public void testEmptyFlagRemovesAnnotatedMethodsControl() throws IOException {
```

Comment:
this test can be removed.

Reply:
Not sure why this test need to be removed?
This test checks if the complete method deletion (with the annotation) does not happen in case of empty flag-name

Reply:
Because the treatment [check](https://github.com/uber/piranha/blob/master/java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java#L857) is already within the conditional [check](https://github.com/uber/piranha/blob/master/java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java#L851) on flagname.  The control [check](https://github.com/uber/piranha/blob/master/java/piranha/src/main/java/com/uber/piranha/XPFlagCleaner.java#L860) is not testing anything new and increases the testing time.

Reply:
with `isTreated=false`, `Piranha:FlagName=""` and fix removed I see `piranha` deletes the annotated method whereas it shouldn't

Reply:
> > Overall, looks good. Can you also add a test case where ` b.putFlag("Piranha:FlagName", "");` is not passed so that we can be sure that an exception is thrown?
> 
> Sorry, did not get you.
> do you mean where the flag config is not present and `PiranhaConfigurationException` is thrown?

Yes.  A test without flagname to check an exception is correctly thrown is needed here.

Reply:
This doesn't contradict whatever I mentioned above.